thrust/thrust/system/detail/sequential/partition.h

```diff
@@ -173,6 +173,11 @@ _CCCL_HOST_DEVICE
                                    ForwardIterator last,
                                    Predicate pred)
 {
+  if (first == last)
+  {
+    return first;
+  }
```

Comment:
@miscco the seg fault in CUB tests happened in sequential (host-side) thrust sort. It seems to be GCC 12 miscompilation, since it only happens in release build when using `thrust::stable_sort` with empty `thrust::host_vector<bool>`. In this case, `stable_partition` is used, which constructs temporary array (no allocation). The seg fault happens when `logical_not` is called on an element of that temporary storage, which shouldn't happen, since it's only called inside the loop. 

I'll try to take a closer look if the changes you've made to pair affect temporary array or it's a bug on GCC end, but for now this fix should unblock you.

Reply:
thats an awesome find thanks a lot